src/sentry/integrations/source_code_management/repository.py

```diff
@@ -9,31 +10,92 @@
 from sentry.integrations.services.repository import RpcRepository
 from sentry.models.identity import Identity
 from sentry.models.repository import Repository
+from sentry.shared_integrations.client.base import BaseApiResponseX
 from sentry.shared_integrations.exceptions import ApiError, IntegrationError
+from sentry.utils import metrics
 
+REPOSITORY_INTEGRATION_CHECK_FILE_METRIC = "repository_integration.check_file.{result}"
+REPOSITORY_INTEGRATION_GET_FILE_METRIC = "repository_integration.get_file.{result}"
 
-class RepositoryMixin:
-    # whether or not integration has the ability to search through Repositories
-    # dynamically given a search query
-    repo_search = False
 
+class RepositoryIntegration(ABC):
```

Comment:
Since this class has an "is-a" relationship to `IntegrationInstallation`, I think it would be good to extend it (i.e., `class RepositoryIntegration(IntegrationInstallation, ABC)`).

Then we won't have to inherit both separately in the subclasses. (That is, instead of writing `class VstsIntegration(IntegrationInstallation, RepositoryIntegration, VstsIssueSync)` it can just be `class VstsIntegration(RepositoryIntegration, VstsIssueSync)`, no longer using the mixin pattern.)